draft-jmiller-json-proof-algorithms.md

```diff
@@ -184,15 +184,163 @@ The Verifier uses only the disclosed payloads and generates or uses the included
+
+### Signing
+
+The Signer's BLS12-381 G2 key pair is used to sign the completed message array input containing the octet strings of the Protected Header and every payload.  The result is a signature octet string that is used as the signed JWP Proof value.
+
+In the implementation, the method used to perform the signing is `blsSign({keyPair, [header, payload1, payload2, ...]})` and returns a binary signature value.
+
+### Proving
+
+The prover must decode the JWP header and payload values in order to generate the identical message array that the signer used.
+
+To generate a JWP for a verifier, the prover must use a cryptographic nonce that is provided by that verifier as input.  This nonce MUST be a 32 byte octet string that the verifier generated from a trusted  source.
+
+The prover also applies selective disclosure preferences by creating an array of indices of which messages in the input array are to be revealed to the verifier.  The revealed indices MUST include the value `0` so that the protected header message is always revealed to the verifier.
+
+The result of creating a proof is an octet string that is used as the verifiable JWP proof value.
+
+In the implementation, the method used to generate the proof is `blsCreateProof({signedProof, publicKey, [header, payload1, payload2, ...], nonce, [0, 2, ...])`.
+
+### Verification
+
+The verifier decodes the JWP header and payload values into a messages array, skipping any non-revealed payloads.  The current BBS implementation embeds the revealed indices into the output proof value so the verification messages array only needs to include the revealed messages.
+
+In the implementation, the method used to verify the proof is `blsVerifyProof({verifyProof, publicKey, [header, payload2, ...] nonce)`.
+
+### JPA Registration
+
+Proposed JWP `alg` value for BBS based on the software implementation is "BBS-X".
+
+### Example
+
+The following example uses the given BLS12-384 key-pair:
+
+Public:
+```json
+[174, 25, 240, 10, 149, 197, 107, 241, 79, 27, 32, 83, 19, 34, 100, 118, 209, 75, 211, 26, 186, 42, 67, 136, 206, 88, 202, 153, 44, 176, 141, 59, 16, 171, 59, 234, 35, 138, 148, 233, 234, 230, 121, 210, 157, 73, 248, 245, 13, 71, 6, 15, 244, 112, 20, 176, 167, 127, 12, 33, 115, 49, 243, 163, 157, 209, 244, 232, 99, 209, 128, 150, 67, 253, 208, 102, 97, 60, 220, 255, 204, 32, 159, 193, 241, 57, 198, 51, 118, 187, 18, 156, 46, 114, 66, 49]
+```
+
+Private:
+```json
+[44, 254, 245, 65, 205, 228, 50, 118, 124, 30, 71, 54, 61, 149, 218, 51, 168, 222, 50, 176, 12, 145, 125, 33, 213, 111, 123, 182, 153, 38, 87, 118]
+```
+
+The protected header used is:
+```json
+{
+  "iss": "https://issuer.tld",
+  "claims": [
+    "family_name",
+    "given_name",
+    "email",
+    "age"
+  ],
+  "typ": "JPT",
+  "alg": "BBS-X"
```

Comment:
ideally we could add a comment here to point to a WG or IETF draft for this value... even pointing to a GitHub repo would be better than just including the magic string here.

Reply:
This value is "registered" below by this document... but the formal language to create the JPA registrations isn't in place yet so it's a bit of a chicken-or-egg at the moment :)